lib/internal/util/parse_args/parse_args.js

```diff
@@ -56,21 +56,22 @@ const {
 
 
 function getMainArgs() {
-  // Work out where to slice process.argv for user supplied arguments.
-
-  // Check node options for scenarios where user CLI args follow executable.
   const execArgv = process.execArgv;
-  if (ArrayPrototypeIncludes(execArgv, '-e') ||
-      ArrayPrototypeIncludes(execArgv, '--eval') ||
-      ArrayPrototypeIncludes(execArgv, '-p') ||
-      ArrayPrototypeIncludes(execArgv, '--print')) {
+
+  const hasEvalOrPrint = execArgv.some(arg =>
+    arg === '-e' || arg === '--eval' ||
+    arg === '-p' || arg === '--print' ||
+    arg.startsWith('--eval=') || arg.startsWith('--print=')
+  );
```

Comment:
This is an improvement, but still not robust – for example, usage like `node -pe "code to eval"` won't match.

This would be better off querying the options directly:

```js
const { getOptionValue } = require('internal/options');

if (getOptionValue('--eval').length !== 0) { ... }
```

`-p`/`--print` sets `--eval` internally, so this should work for both cases.